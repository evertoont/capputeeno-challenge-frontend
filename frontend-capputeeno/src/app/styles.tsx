"use client";

import styled from "styled-components";

export const PageWrapper = styled.main`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const DefaultPageLayout = styled.div`
  min-height: calc(100vh - 100px);
  padding: 34px 50px;
  background-color: ${({ theme }) => theme.colors.primary};

  @media (min-width: ${({ theme }) => theme.breakpoints.desktop}) {
    padding: 34px 160px;
  }
`;

export const Divider = styled.span`
  width: 100%;
  height: 1px;
  margin: 8px auto;
  padding: 0px;
  background: ${({ theme }) => theme.colors.shapes};
`;
