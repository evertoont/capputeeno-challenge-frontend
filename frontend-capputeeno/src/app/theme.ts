export const colors = {
  logo: '#5D5D6D',
  secondary: '#F3F5F6',
  textDark: '#737380',
};

export const theme = {
  colors
}