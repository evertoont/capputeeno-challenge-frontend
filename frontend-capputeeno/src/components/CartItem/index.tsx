import { ProductInCart } from "@/types/products";
import {
  CartItemContainer,
  ProductDeleteButton,
  ProductImage,
  ProductInfoContainer,
  ProductInfoDescription,
  ProductInfoFooter,
  ProductInfoHeader,
  ProductInfoPrice,
  ProductInfoTitle,
  SelectQuantity,
} from "./styles";
import { DeleteIcon } from "../icons/delete";
import { formatPrice } from "@/utils/produtcs";
import { useProductSettings } from "@/hooks/useProductSettings";

interface CartItemProps {
  productInfo: ProductInCart;
  onRemoveProduct: (productId: string) => void;
  onUpdateQuantity: (productId: string, quantity: number) => void;
}
export function CartItem(props: CartItemProps) {
  const { productInfo, onRemoveProduct, onUpdateQuantity } = props;
  const { maxQuantityItems } = useProductSettings();

  const handleUpdateQuantity = (
    event: React.ChangeEvent<HTMLSelectElement>
  ) => {
    onUpdateQuantity(productInfo.id, Number(event.target.value));
  };

  return (
    <CartItemContainer>
      <ProductImage src={productInfo.image_url} />
      <ProductInfoContainer>
        <ProductInfoHeader>
          <ProductInfoTitle>{productInfo.name}</ProductInfoTitle>
          <ProductDeleteButton onClick={() => onRemoveProduct(productInfo.id)}>
            <DeleteIcon />
          </ProductDeleteButton>
        </ProductInfoHeader>

        <ProductInfoDescription>
          {productInfo.description}
        </ProductInfoDescription>

        <ProductInfoFooter>
          <SelectQuantity
            value={productInfo.quantity}
            onChange={handleUpdateQuantity}
          >
            {Array.from({ length: maxQuantityItems }, (_, index) => (
              <option key={index} value={index + 1}>
                {index + 1}
              </option>
            ))}
          </SelectQuantity>
          <ProductInfoPrice>
            {formatPrice(productInfo.price_in_cents)}
          </ProductInfoPrice>
        </ProductInfoFooter>
      </ProductInfoContainer>
    </CartItemContainer>
  );
}
