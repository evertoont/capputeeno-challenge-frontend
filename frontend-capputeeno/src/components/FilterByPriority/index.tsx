"use client";

import { useState } from "react";
import { ArrowIcon } from "../icons/arrow";
import {
  FilterContainerPriority,
  FilterOption,
  FilterWrapperOptions,
  MainButtonSelected,
} from "./styles";
import { priorityFilterItems } from "@/utils/filters";

export function FilterByPriority() {
  const [isFilterOpen, setIsFilterOpen] = useState(false);

  const handleOpenFilter = () => {
    setIsFilterOpen(!isFilterOpen);
  };

  return (
    <FilterContainerPriority>
      <MainButtonSelected onClick={handleOpenFilter}>
        Organizar por
        <ArrowIcon />
      </MainButtonSelected>

      {isFilterOpen && (
        <FilterWrapperOptions>
          {priorityFilterItems.map((item) => {
            return (
              <FilterOption onClick={handleOpenFilter} key={item.type}>
                {item.label}
              </FilterOption>
            );
          })}
        </FilterWrapperOptions>
      )}
    </FilterContainerPriority>
  );
}
