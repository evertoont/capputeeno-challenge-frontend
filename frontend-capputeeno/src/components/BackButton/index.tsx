import { useRouter } from "next/navigation";
import { ButtonText, ContainerButton } from "./styles";
import { BackIcon } from "../icons/back";

interface BackButtonProps {
  navigateTo: string;
}
export function BackButton(props: BackButtonProps) {
  const { navigateTo } = props;
  const router = useRouter();

  const handleBackButton = () => {
    router.push(navigateTo);
  };

  return (
    <ContainerButton onClick={handleBackButton}>
      <BackIcon />
      <ButtonText>Voltar</ButtonText>
    </ContainerButton>
  );
}
