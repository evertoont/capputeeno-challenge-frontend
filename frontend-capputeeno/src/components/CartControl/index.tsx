"use client";

import { useRouter } from "next/navigation";
import { CartIcon } from "../icons/cart";
import { CartCount, CartContainer } from "./styles";
import { useReadLocalStorage } from "usehooks-ts";
import { ProductInCart } from "@/types/products";

export function CartControl() {
  const router = useRouter();
  const productCartList = useReadLocalStorage<ProductInCart[]>("product-cart");

  const existProductCart = productCartList && productCartList.length > 0;

  const handleClickCart = () => {
    router.push("/cart");
  };

  return (
    <CartContainer onClick={handleClickCart}>
      <CartIcon />
      {existProductCart && <CartCount>{productCartList?.length}</CartCount>}
    </CartContainer>
  );
}
