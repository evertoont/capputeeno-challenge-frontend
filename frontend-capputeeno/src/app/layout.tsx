import "./globals.css";
import "react-toastify/dist/ReactToastify.css";
import "react-loading-skeleton/dist/skeleton.css";

import type { Metadata } from "next";
import { DefaultProviders } from "@/components/DefaultProviders";
import { Header } from "@/components/Header";
import { Saira } from "next/font/google";
import { ToastContainer } from "react-toastify";

const saira = Saira({
  weight: ["300", "400", "500", "600"],
  subsets: ["latin"],
});

export const metadata: Metadata = {
  title: "Capputeeno - Challenge",
  description: "Capputeeno - Challenge E-commerce",
};

export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body className={saira.className}>
        <ToastContainer
          position="top-center"
          autoClose={5000}
          hideProgressBar={false}
          newestOnTop={false}
          closeOnClick
          rtl={false}
          pauseOnFocusLoss
          draggable
          pauseOnHover
          theme="light"
        />
        <DefaultProviders>
          <Header />
          {children}
        </DefaultProviders>
      </body>
    </html>
  );
}
