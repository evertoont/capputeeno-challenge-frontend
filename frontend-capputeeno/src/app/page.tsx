import Image from "next/image";
import styles from "./page.module.css";
import { DefaultPageLayout, PageWrapper } from "./styles";
import { FilterBar } from "@/components/FilterBar";

export default function Home() {
  return (
    <DefaultPageLayout>
      <PageWrapper>
        <FilterBar />
      </PageWrapper>
    </DefaultPageLayout>
  );
}
