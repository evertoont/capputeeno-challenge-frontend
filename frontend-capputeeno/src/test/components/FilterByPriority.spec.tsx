import { theme } from "@/app/theme";
import { FilterByPriority } from "@/components/FilterByPriority";
import { fireEvent, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

describe("FilterByPriority", () => {
  it("should render FilterByPriority component", () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <FilterByPriority />
      </ThemeProvider>
    );

    expect(getByTestId("FilterByPriority")).toMatchSnapshot();
  });

  it("should render FilterByPriority component with options", () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <FilterByPriority />
      </ThemeProvider>
    );

    const filterByPriority = getByText("Organizar por");

    fireEvent.click(filterByPriority, "click");

    expect(getByText("Novidades")).toBeInTheDocument();
    expect(getByText("Preço: Maior - menor")).toBeInTheDocument();
    expect(getByText("Preço: Menor - maior")).toBeInTheDocument();
    expect(getByText("Mais vendidos")).toBeInTheDocument();
  });

  it("should render FilterByPriority component with options and click in one of them", () => {
    const { getByText, queryByText } = render(
      <ThemeProvider theme={theme}>
        <FilterByPriority />
      </ThemeProvider>
    );

    const filterByPriority = getByText("Organizar por");

    fireEvent.click(filterByPriority, "click");

    expect(getByText("Novidades")).toBeInTheDocument();

    const filterByPriorityOption = getByText("Novidades");

    fireEvent.click(filterByPriorityOption, "click");

    expect(queryByText("Novidades")).not.toBeInTheDocument();
  });
});
