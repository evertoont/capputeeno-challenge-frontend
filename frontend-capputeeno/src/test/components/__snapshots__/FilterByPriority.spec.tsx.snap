// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByPriority should render FilterByPriority component 1`] = `
<div
  class="sc-bcPKhP gckuTJ"
  data-testid="FilterByPriority"
>
  <button
    class="sc-grXZZQ jFNDq"
  >
    Organizar por
    <svg
      fill="none"
      height="6"
      viewBox="0 0 10 6"
      width="10"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 1L5 5L9 1"
        stroke="#737380"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </button>
</div>
`;
